geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -2033,6 +2035,29 @@ protected int getRegionSize() {
   }
 
   /**
+   * Returns the size in this region.
+   *
+   * This is used in a transaction to find the size of the region on the transaction hosting node.
+   *
+   * @param target the host of the transaction TXState
+   * @return the number of entries in this region
+   */
+  public int getRegionSize(DistributedMember target) {
```

Comment:
Otherwise we need to handle it based on different region types using instanceof. I think this is a clean way to do this.

Reply:
It certainly is convenient, but that is how we have gotten to the state we are in with LocalRegion today. Personally, I would prefer some contained ugliness (like instanceof statements) elsewhere than expanding the interface for already bloated region classes. I also concur with @agingade on messaging logic in LocalRegion instead of in a distributed region flavor.